nemo/core/connectors/save_restore_connector.py

```diff
@@ -0,0 +1,325 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2015 and onwards Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pickle
+import shutil
+import tarfile
+import tempfile
+import uuid
+from os import path
+from typing import Optional, Union
+
+import torch
+from omegaconf import DictConfig, OmegaConf
+from omegaconf.omegaconf import open_dict
+
+from nemo.utils import app_state, logging, model_utils
+from nemo.utils.app_state import AppState
+from nemo.utils.model_utils import import_class_by_path
+
+
+class SaveRestoreConnector:
+    def _default_save_to(self, model, save_path: str):
+        """
+			Saves model instance (weights and configuration) into .nemo file.
+			You can use "restore_from" method to fully restore instance from .nemo file.
+
+			.nemo file is an archive (tar.gz) with the following:
+				model_config.yaml - model configuration in .yaml format. You can deserialize this into cfg argument for model's constructor
+				model_wights.chpt - model checkpoint
+
+			Args:
+				save_path: Path to .nemo file where model instance should be saved
+
+		"""
+        app_state = AppState()
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            config_yaml = path.join(tmpdir, app_state.model_config_yaml)
+            model_weights = path.join(tmpdir, app_state.model_weights_ckpt)
```

Comment:
I am really not sure if those two "constants" should be attributes of app_state.

Why not make them attributes of the Connector class?

Reply:
That should work now that we have the connector class :)

Reply:
Done.